fluentd/lib/filter_elasticsearch_genid_ext/lib/filter_elasticsearch_genid_ext.rb

```diff
@@ -0,0 +1,47 @@
+require 'securerandom'
+require 'base64'
+require 'fluent/filter'
+
+module Fluent
+  class ElasticsearchGenidExtFilter < Filter
+    Fluent::Plugin.register_filter('elasticsearch_genid_ext', self)
+
+    config_param :hash_id_key, :string, :default => '_hash'
+    config_param :alt_key, :string, default: ""
+
+    def initialize
+      super
+    end
+
+    def configure(conf)
+      super
+      @alt_key = conf['alt_key']
```

Comment:
`config_param` already does this - it creates an instance variable called `@alt_key` and sets it to the configured value.  You can omit the `configure` method too

Reply:
Thanks!  Done.